### DISCUSSION TITLE
EleksMaker Stops and Just Makes Lines

### DISCUSSION THREAD

**Discussion by 3dillon** [arkypita/LaserGRBL#947] [Q&A]
Hello,

I am currently new to this and am running an Eleksmaker that I recently acquired from someone who passed away. I believe it is a 2.5 W laser and it has three axis setup on it.
I have been able to successfully use the laser using LaserGRBL V3.1.2 software. I initially tried EleksCAM but without any success (the port will say it is connected but the software does not seem to be able to control the laser at all). When about a quarter of the way through a job using LaserGRBL the system suddenly stops and just goes back and forth making lines instead of continuing the job.

Do you have any advice on how to make this stop? 

Thanks in advance.

**Comment by StuartB4** [comment]
Does it do it with every image you try?

Which version of GRBL does it have. You can see that by starting LaserGRBL and click on Connect, then look in the window on the left underneath the Progress bar.

Have you tried all the different methods of engraving like Line2Line, 1Bit Dithering and Vectorize.
If so, it is happening in all modes. 

When you load an image in what settings are you using, also the settings in Target Image settings.

**Comment by 3dillon** [comment]
Hello,
I appreciate your response. I have tried a handful of images and it seems to occur for each one. I have tried .Jpeg, .png and bitmaps. 
The first message bar that appears after connection is green and states: Grbl 0.9i ['$' for help]. Is 0.9i the GRBL version?
I have only tried Line To Line Tracing. By the looks of the preview image in the editor, this method is the only of the four that seems to keep its clarity of the image. Should I attempt the others?
I have tested both horizontal and diagonal directions at a quality of 5.000 Lines/mm. I simply left the brightness, contrast and whiteclip ad defualt. Resize : Smooth. Grayscale: SimpleAverage.
For target image:
Engraving speed has been between 1000 mm/min to 1500 mm/min.
Laser ON: M3, Laser OFF: M5
S-MIN: 0, S-MAX: Has been tested between 100 to 255 with 130ish being best.
Image size: W 115.0 and H 161.0

Thanks again!

**Comment by 3dillon** [comment]
![image](https://user-images.githubusercontent.com/61444196/78626267-3c2b0c80-785c-11ea-9baa-09d57fa16f07.png)
 Here is an example of the line that it suddenly starts cutting instead of continuing with the project.

**Comment by arkypita** [comment]
I assume that the gcode commands generated by LaserGRBL are right, and the preview does show the image in a correct way. Is it?

So this should be an hardware or firmware issue.

With hardware I mean something that limit the movement of the laser head in its travel. Are you able to move the laser on all surface using jog arrows in the bottom left area of LaserGRBL interface? Is the movement fluent? Does it respond correctly to jog directions across the surface?

With firmware I mean that Grbl (that is a program running inside the chip that is in the engraver board) could be misconfigured or wrongly installed.

I have some question:
- In the picture you upload, what was the direction of engraving? Does it always make diagonal "cutting" regardless of the direction of engraving, or does it make diagonal line if you draw diagonally and horizontally if you select horizontal engraving?
- How is the behavior of the reference cross showing the position of the laser on the preview in LaserGRBL? Does it move in line with what the laser is physically doing (ie goes up and down diagonally following the cutting line) or does it continue to follow the original design?
- Can you upload a picture of your engraver and a picture (or ctrl-c ctrl-v) of your grbl settings (menu grbl -> configure grbl). Please extend the window to see all the settings.
